### DISCUSSION TITLE
🤯🎉 Bump v2.0.0 and rename to Lobe Theme, add several theme customize features

### DISCUSSION THREAD

**Discussion by canisminor1990** [lobehub/sd-webui-lobe-theme#143] [Announcements]
> 📦 After **Version 2.0.0** Kitchen theme was renamed to **Lobe Theme**. The legacy version can be accessed at [sd-webui-kitchen-theme-legacy](https://github.com/canisminor1990/sd-webui-kitchen-theme-legacy)

<br/>

## ✨ Features

- [x] 🌗 Supports light and dark themes, with the ability to quickly switch in the navigation bar
- [x] 🌈 Supports custom theme colors and neutral colors, with the option to customize the logo
- [x] 🪄 Supports one-click formatting of the prompt, with a simple tag editor provided
- [x] 🎛️ Highly customizable sidebar, with a quick settings sidebar on the left and a model sidebar on the right
- [x] 🖼️ Adjustable canvas ratio, ensuring that generated images are always displayed at the top
- [x] 📱 Mobile-friendly, with partial optimization for mobile screens
- [ ] 📝 Syntax highlighting in the prompt input box
- [ ] 🆗 Multilingual support with i18n

## 🤯 Usage

![feat_thememode.webp](https://raw.githubusercontent.com/canisminor1990/sd-webui-lobe-theme/main/docs/feat_thememode.webp)

#### Light and Dark Themes

> 👉 Tip: You can quickly switch between light and dark themes in the upper-right corner of the navigation bar.

The current theme supports both light and dark themes. If you want to force dark mode, use the `--theme=dark` argument to launch your WebUI. For example, on Windows, your `webui-user.bat` should include:

```shell
set COMMANDLINE_ARGS= --theme=dark
```

Alternatively, you can switch directly through URL Props:

```shell
http://localhost:7860/?__theme=light
http://localhost:7860/?__theme=dark
```

![feat_theme_modify.webp](https://raw.githubusercontent.com/canisminor1990/sd-webui-lobe-theme/main/docs/feat_theme_modify.webp)

#### Theme Customization

> 👉 Tip: Click the `⚙` icon in the upper-right corner to open the settings panel. The current available settings are as follows:

- **Theme**
  - Primary color: Currently offers `13` theme color combinations
  - Neutral color: Currently offers `6` different grayscale color combinations
  - Logo type: `Lobe`, `Kitchen`, `Custom`
    - Custom logo: Supports `img url`, `base64`, and `emoji`. When a single emoji is entered, it will be automatically replaced with 3D Fluent Emoji.
    - Custom title: Customize the site name.

![feat_sidebar.webp](https://raw.githubusercontent.com/canisminor1990/sd-webui-lobe-theme/main/docs/feat_sidebar.webp)

#### Sidebar Customization

> 👉 Tip: Click the `⚙` icon in the upper-right corner to open the settings panel. The current available settings are as follows:

- **Prompt Textarea**
  - Display mode: `scroll fixed height` | `resizable by text lines`
- **Sidebar**
  - Default expand: `true`
  - Display mode: `fixed` | `float`
  - Default width: `280`
- **ExtraNetwork Sidebar**
  - Enable: `true`
  - Default expand: `true`
  - Display mode: `fixed` | `float`
  - Default width: `340`
  - Default card size: `86`

<details>
<summary><kbd>Recommended System Settings</kbd></summary>

#### Extra-Networks

- thumbs
- width: 86
- height: 128

<br/>

#### Quick-Setting

```txt
sd_model_checkpoint, sd_vae, CLIP_stop_at_last_layers, img2img_background_color, img2img_color_correction, samples_save, samples_format, grid_save, return_grid,  n_rows, live_previews_enable, show_progress_every_n_steps, live_preview_refresh_period
```

</details>

![feat_mobile_friendly.webp](https://raw.githubusercontent.com/canisminor1990/sd-webui-lobe-theme/main/docs/feat_mobile_friendly.webp)

#### Mobile Adaptation

Partial convenience adaptation design has been completed for mobile devices, including breadcrumb foldable navigation bar, sidebar adaptation, etc. However, due to the high complexity and fixed values of the stable diffusion interface, it is difficult to ensure the same user experience as the desktop version. Feedback is welcome for more ideas.

**Comment by amimi818** [comment]
真是颠覆性的升级，还在慢慢体验中：
1. 生成按钮/结果区和配置区分开不同列很好，生成时不用上下拖动页面了，但带来的界面调整似乎还要适应一下，尤其原来“生成”按钮下面的一些图标及提示词模版（styles）选择框放到左边，进一步占用了左部的宝贵空间。
可以考虑能否将它们仍放在“生成”按钮下面，或者做一个选项，让用户自己选择；再或者把图标和styles放到同一行，几个图标占用一行有点浪费；
2. 升级后不知怎么把我安装的没有启用的插件也全部默认启用了，似乎没有继承原来的启用设置；
3. 双语情况下，顶部菜单中英文同行显示不太友好，太宽了；
4. 浅色主题下，一些矢量图标似乎没有适配；
5. 似乎比原来有点卡。。。

此外，右上角的快捷链接能否支持用户自定义，或者至少把[Cheat Sheet]放上去（感觉它值得在那儿拥有一个位置，下面虽然已经有链接，还是在上面更方便）。

再次感谢作者的杰出工作，让我们能在gradio框架下还可以享用如此美观舒适的界面。

<img width="708" alt="image" src="https://github.com/canisminor1990/sd-webui-lobe-theme/assets/133093934/1cda345f-5d0d-4acb-bc54-2bd8d3061224">

**Reply by canisminor1990** [reply]
正在慢慢修复 bug 感谢体验建议😛

**Reply by Crazypeter0801** [reply]
请问左边的操作如何自定义？Mac系统 只有选择大模型一个操作

**Comment by amimi818** [comment]
晚上在家又体验了一下升级后的kitchen（lobehub），没想到这么快就又更新了，补充一点新感受：
1. 增加了传统布局，很好，把选择权交给用户；
2. 主题颜色虽然增加了很多选项，但无论我怎么配置，也无法调出原来kitchen的主题色了。似乎背景更深了，字体更亮了。有点接近原生1111的配色。而我喜欢kitchen的一大原因就是那种不同于原生页面的灰色系配色。希望能保留kitchen1.0的配色选择（包括字体颜色亮度）；
3. 右侧的扩展sidebar不知道是不是我的个案，悬浮模式下，当我鼠标移到屏幕最右侧时，sidebar出不来，鼠标必须稍稍往左收回来一点才可以。左侧的快捷sidebar就没有这个问题，不知道是不是和右侧有上下滚动条有关（kitchen1.0没有这个问题）；
4. Generate按钮的动态发光效果能否设置成可取消；
5. Seed的随机和Reset两个按钮调整成和Seed输入框等宽有点不习惯，最好是还改为小按钮形式；
6. 左侧sidebar悬出的时候，部分主页面的内容（如采样方法的标签文字、图生图中的Resise Mode等）居然是一直在顶层。

先写这么多，都是一些碎碎念，也不发issue了，不知道作者会不会看讨论区的建议，真心希望这个主题能越做越好。。。

![Quicker_20230628_232411](https://github.com/canisminor1990/sd-webui-lobe-theme/assets/133093934/088e5e25-6083-4b11-bb70-9f7fd8b12606)

**Reply by canisminor1990** [reply]
感谢建议，正在慢慢优化🤗

**Reply by canisminor1990** [reply]
发光效果和 kitchen legacy 颜色配置现在已经添加入设置中，主题色和中性色第二个就是

**Comment by amimi818** [comment]
感谢！更新迭代好快，这么快就修复了这么多问题，注意休息哈。
